### ISSUE TITLE
toggle smooth scroll on/off for select input devices

### ISSUE THREAD

**Issue by mfrisina6600** [Caldis/Mos#682]
loving this application but I'm in a situation where I only want it to work for my mouse, not my pen display's drawing pen. I'm a digital artist and I sadly can't use Mos because whenever I swap from scrolling with my Logitech mouse to my Wacom pen, the scrolling gets super wonky while using the pen. I wish I could tell Mos to specifically enable smooth scrolling while using the mouse vs. the pen. I'd even pay for a feature like this tbh

**Comment by moeleak** [comment]
Yeah. I agree with you. I got the same situation. I'd like to pay for this feature

**Comment by yqmarkqq** [comment]
Samy here

**Comment by hacker-DOM** [comment]
=== Solution

Use Mos just for smooth scrolling and use Karabiner for setting Vertical reverse.

<img width="370" alt="Image" src="https://github.com/user-attachments/assets/40235e60-e68b-402b-b9ed-eeb323be0ab4" />

<img width="935" alt="Image" src="https://github.com/user-attachments/assets/60aad50b-9aac-4f85-8ace-af81476ff32a" />

**Comment by mfrisina6600** [comment]
> === Solution
> 
> Use Mos just for smooth scrolling and use Karabiner for setting Vertical reverse.
> 
> <img alt="Image" width="370" src="https://private-user-images.githubusercontent.com/18601956/443066194-40235e60-e68b-402b-b9ed-eeb323be0ab4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxNzI5NDMsIm5iZiI6MTc1NTE3MjY0MywicGF0aCI6Ii8xODYwMTk1Ni80NDMwNjYxOTQtNDAyMzVlNjAtZTY4Yi00MDJiLWI5ZWQtZWViMzIzYmUwYWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE0VDExNTcyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkYjEzNmY2NGI1ZmU2MDAxODRjOWE3NTg2OTEyOTZlMzdjY2I3ODUyNzhiM2FlODM2MjQ0MGI0MDk1ZmEwZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EAPE_1qVtvqUMjJPmmGs_8VwGJmEqMFp3InuJaxiNwE"> <img alt="Image" width="935" src="https://private-user-images.githubusercontent.com/18601956/443066011-60aad50b-9aac-4f85-8ace-af81476ff32a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxNzI5NDMsIm5iZiI6MTc1NTE3MjY0MywicGF0aCI6Ii8xODYwMTk1Ni80NDMwNjYwMTEtNjBhYWQ1MGItOWFhYy00Zjg1LThhY2UtYWY4MTQ3NmZmMzJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE0VDExNTcyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ4NDgyNzMwYzY3NTVkMmY0OTBkOGMyM2U4M2Y4MzZlMGE5ZDQ5NjM5N2MwZDM1N2FiYjc5NmJjY2E1YjJhZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VbjF9qRiV8ST_9uoe5hgAzhyMBnsgrQ2OMopTJhm2Ug">

Sorry I’m confused how this is a solution… the issue wasn’t about the scroll direction.

**Comment by hacker-DOM** [comment]
Ak ok, for me it was about scroll direction. Could apply to both I prepose

**Comment by goonzoid** [comment]
> Ak ok, for me it was about scroll direction. Could apply to both I prepose

Unfortunately not, since Karabiner has no concept of smooth scrolling afaik.

I also need this feature. Would be happy to implement and contribute, but would rather not put the effort in up front if it's unlikely to be merged. Would be great to get a sign from the maintainer that it would be welcome

**Comment by lasergyro** [comment]
I used AI to implement a per-device Mos toggle here: https://github.com/lasergyro/Mos/tree/feat/per-device-toggle
It works for me, and likely you can ask AI to guide you through compiling it and enabling it on your system.
